types/deneb/block_contents.yaml

```diff
@@ -1,37 +1,36 @@
 Deneb:
+  KZGProofs:
+    type: array
+    items:
+      $ref: '../primitive.yaml#/KZGProof'
+    minItems: 0
+    maxItems: 6
+
+  Blobs:
+    type: array
+    items:
+      $ref: "../primitive.yaml#/Blob"
+    minItems: 0
+    maxItems: 6
+  
   BlockContents:
     type: object
     description: "The required object for block production according to the Deneb CL spec."
     properties:
       block:
         $ref: "./block.yaml#/Deneb/BeaconBlock"
-      blob_sidecars:
-        $ref: "./blob_sidecar.yaml#/Deneb/BlobSidecars"
-
-  BlindedBlockContents:
-    type: object
-    description: "The required object for blinded block production according to the Deneb CL spec."
-    properties:
-      blinded_block:
-        $ref: "./block.yaml#/Deneb/BlindedBeaconBlock"
-      blinded_blob_sidecars:
-        $ref: "./blob_sidecar.yaml#/Deneb/BlindedBlobSidecars"
+      kzg_proofs:
```

Comment:
if we are assuming statelessness, are we requiring here for the beacon to build inclusion proofs for the blobs before transmitting them?

Reply:
The validator can compute them on its own, the inclusion proof can be computed solely from `block`.

Reply:
correct however what i was clarifying that since validator publishes signed block contents, the intercepting beacon is would be building the proofs and publishing blocks  (which is fine or may be good as well that while validator is signing the beacon can build the proofs and patch them in as validator starts publish flow although just a matter of 20-30ms as such)

(this comment is about SignedBlockContents)

Reply:
I was thinking in implementation, a beacon node could start building the inclusion proofs and cache them when it builds a block. But should also have the ability to compute them on the fly (on a cache miss). So in the case where you publish through the BN building the block it'd be faster.

Reply:
right, lodestar will soon be transition to SSZ now lol